app/services/SessionService.scala

```diff
@@ -221,26 +222,26 @@ class SessionService @Inject()(metrics: ApplicationMetrics, sessionCache: GmpSes
     })
   }
 
-  def fetchScenario()(implicit request: Request[_], hc: HeaderCarrier): Future[Option[String]] = {
+  def fetchScenario()(implicit request: Request[_], hc: HeaderCarrier ,gmpsessionCache: GmpSessionCache): Future[Option[String]] = {
```

Comment:
Is there a reason why you decided to make GmpSessionCache implicit everywhere instead of injecting it at class level once and use it everywhere????

Reply:
moved to class level